CI: Gateway API Conformance Test

### Test Name

TestConformance/UDPRouteWeightedRouting

### Test Failure Output

UDPRouteWeightedRouting

```shell
    --- FAIL: TestConformance/UDPRouteWeightedRouting/UDP_traffic_should_be_distributed_across_the_weighted_backends (0.19s)

    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.740047848Z: UDP weighted distribution attempt 1/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:33553->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.765246154Z: UDP weighted distribution attempt 2/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:54931->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.785093988Z: UDP weighted distribution attempt 3/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:58661->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.806699621Z: UDP weighted distribution attempt 4/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:40636->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.826511741Z: UDP weighted distribution attempt 5/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:51114->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.853252433Z: UDP weighted distribution attempt 6/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:59278->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.868873777Z: UDP weighted distribution attempt 7/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:60154->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.889053002Z: UDP weighted distribution attempt 8/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:54419->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.90544415Z: UDP weighted distribution attempt 9/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:48891->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:79: 2026-07-03T15:54:02.917005687Z: UDP weighted distribution attempt 10/10 failed: error while sending requests: reading UDP echo response: read udp 172.18.0.1:56776->172.18.255.196:5300: read: connection refused
    udproute-weighted-routing.go:84: UDP weighted distribution did not converge within tolerance
```

TestConformance/TCPRouteWeightedRouting

```shell
    --- FAIL: TestConformance/TCPRouteWeightedRouting/TCP_traffic_should_be_distributed_across_the_weighted_backends_and_skip_the_zero-weighted_backend (0.08s)

    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.767396036Z: TCP weighted distribution attempt 1/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.776291244Z: TCP weighted distribution attempt 2/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.783931725Z: TCP weighted distribution attempt 3/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.792733235Z: TCP weighted distribution attempt 4/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.801415466Z: TCP weighted distribution attempt 5/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.809255514Z: TCP weighted distribution attempt 6/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.816721431Z: TCP weighted distribution attempt 7/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.823150794Z: TCP weighted distribution attempt 8/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.829994698Z: TCP weighted distribution attempt 9/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:70: 2026-07-03T15:53:26.838766187Z: TCP weighted distribution attempt 10/10 failed: error while sending requests: dialing TCP 172.18.255.196:9300: dial tcp 172.18.255.196:9300: connect: connection refused
    tcproute-weighted-routing.go:75: TCP weighted distribution did not converge within tolerance
```

### CI Build Link

https://github.com/cilium/cilium/actions/runs/28670450509/job/85032512215

### Relevant Log Output

```shell

```

### Sysdump / Zip File

_No response_

--- a/.github/workflows/conformance-gateway-api.yaml
+++ b/.github/workflows/conformance-gateway-api.yaml
@@ -199,9 +199,9 @@ jobs:
         run: |
           echo sha=${{ steps.default_vars.outputs.sha }} >> $GITHUB_OUTPUT
 
-          SKIPPED_TESTS="TCPRouteWeightedRouting,UDPRouteWeightedRouting"
+          SKIPPED_TESTS=""
           if [ "${{ matrix.conformance-profile }}" == "true" ]; then
-            SKIPPED_TESTS+=",MeshConsumerRoute,HTTPRouteListenerPortMatching"
+            SKIPPED_TESTS="MeshConsumerRoute,HTTPRouteListenerPortMatching"
           fi
 
           # The conformance-profile jobs consume SKIPPED_TESTS through the
@@ -215,7 +215,10 @@ jobs:
           # flows through make's $(GATEWAY_TEST_FLAGS), where a trailing '$'
           # breaks shell quoting; none of the skipped names is a prefix of
           # another conformance test, so the alternation cannot over-match.
-          SKIPPED_TESTS_REGEX="TestConformance/(${SKIPPED_TESTS//,/|})"
+          SKIPPED_TESTS_REGEX=""
+          if [ -n "${SKIPPED_TESTS}" ]; then
+            SKIPPED_TESTS_REGEX="TestConformance/(${SKIPPED_TESTS//,/|})"
+          fi
 
           CILIUM_INSTALL_DEFAULTS="${{ steps.default_vars.outputs.cilium_install_defaults }} \
             --helm-set=kubeProxyReplacement=true \

--- a/Documentation/network/servicemesh/gateway-api/gateway-api.rst
+++ b/Documentation/network/servicemesh/gateway-api/gateway-api.rst
@@ -23,7 +23,7 @@ See the `Gateway API site <https://gateway-api.sigs.k8s.io/>`__ for more details
 Cilium Gateway API Support
 ##########################
 
-Cilium supports Gateway API v1.6.0 for below resources, all the Core conformance
+Cilium supports Gateway API v1.6.1 for below resources, all the Core conformance
 tests are passed.
 
 - `GatewayClass <https://gateway-api.sigs.k8s.io/reference/api-types/gatewayclass/>`_
@@ -77,7 +77,7 @@ Cilium's Gateway API features:
    access-logs
    listenerset
 
-More examples can be found in the `upstream repository <https://github.com/kubernetes-sigs/gateway-api/tree/v1.3.0/examples/standard>`_.
+More examples can be found in the `upstream repository <https://github.com/kubernetes-sigs/gateway-api/tree/v1.6.1/examples/standard>`_.
 
 Troubleshooting
 ###############

--- a/Documentation/network/servicemesh/gateway-api/installation.rst
+++ b/Documentation/network/servicemesh/gateway-api/installation.rst
@@ -6,7 +6,7 @@ Prerequisites
   replacement <kubeproxy-free>`.
 * Cilium must be configured with the L7 proxy enabled using ``l7Proxy=true``
   (enabled by default).
-* The below CRDs from Gateway API v1.6.0 ``must`` be pre-installed.
+* The below CRDs from Gateway API v1.6.1 ``must`` be pre-installed.
   Please refer to these `docs <https://gateway-api.sigs.k8s.io/guides/getting-started/introduction/#installing-gateway-api>`_
   for installation steps. Alternatively, the below snippet could be used.
 
@@ -22,13 +22,13 @@ Prerequisites
 
     .. code-block:: shell-session
 
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_gateways.yaml
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_httproutes.yaml
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_referencegrants.yaml
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_grpcroutes.yaml
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_backendtlspolicies.yaml
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_tlsroutes.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_gateways.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_httproutes.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_referencegrants.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_grpcroutes.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_backendtlspolicies.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_tlsroutes.yaml
 
   If you wish to use the ListenerSet, TCPRoute, or UDPRoute functionality, you
   also need to install the related CRDs. For each CRD that is not installed,
@@ -44,19 +44,19 @@ Prerequisites
 
     .. code-block:: shell-session
 
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_listenersets.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_listenersets.yaml
 
   TCPRoute:
 
     .. code-block:: shell-session
 
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_tcproutes.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_tcproutes.yaml
 
   UDPRoute:
 
     .. code-block:: shell-session
 
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.0/config/crd/standard/gateway.networking.k8s.io_udproutes.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v1.6.1/config/crd/standard/gateway.networking.k8s.io_udproutes.yaml
 
 * By default, the Gateway API controller creates a service of LoadBalancer type,
   so your environment will need to support this. Alternatively, since Cilium 1.16+,

--- a/go.mod
+++ b/go.mod
@@ -137,8 +137,8 @@ require (
 	k8s.io/metrics v0.36.2
 	k8s.io/utils v0.0.0-20260617174310-a95e086a2553
 	sigs.k8s.io/controller-runtime v0.24.1
-	sigs.k8s.io/gateway-api v1.6.0
-	sigs.k8s.io/gateway-api/conformance v1.6.0
+	sigs.k8s.io/gateway-api v1.6.1
+	sigs.k8s.io/gateway-api/conformance v1.6.1
 	sigs.k8s.io/mcs-api v0.5.2
 	sigs.k8s.io/mcs-api/conformance v0.5.2
 	sigs.k8s.io/mcs-api/controllers v0.5.2

